src/daemon.rs

```diff
@@ -147,8 +156,12 @@ async fn main() -> Result<(), Box<dyn Error>> {
     };
 
     let mut hotkeys = load_config();
+
     // Escalating back to root after reading config file.
-    seteuid(0);
+    setinitgroups(&root_user, root_resuid.real.as_raw());
```

Comment:
I think this needs to be in reverse order. First you need to restore effective UID 0 for being able to restore effective GID and auxiliary groups. The way this currently is I guess both setinigroups and setegid will fail with EPERM.

Reply:
Ah I see you fixed that in a follow-up commit. Just ignore this then.

Reply:
> I think this needs to be in reverse order. First you need to restore effective UID 0 for being able to restore effective GID and auxiliary groups. The way this currently is I guess both setinigroups and setegid will fail with EPERM.

Yep, I experienced `EPERM` errors when re-escalating the EGID and initgroups before. I just rearranged the function calls and it's solved now.